explorer/src/components/JupiterSwapModal.tsx

```diff
@@ -0,0 +1,562 @@
+					mint: new PublicKey(tokenAddress),
+				},
+				);
+				const balance = info.value[0].account.data.parsed.info.tokenAmount.uiAmount;
+				if (type === "pay") {
+					setBalance(balance);
+					if ((balance) - 0.00001 > 0) {
+						setInputBalance(balance - 0.00001);
+					} else {
+						setInputBalance(0);
+					}
+				} else if (type === "receive") {
+					setRecieveTokenBalance(balance);
+				}
+			} catch (error) {
+				console.log(error);
+			}
+		}
+	}
+
+	useEffect(() => {
+		setTokenLoading(true);
+		fetch(TOKEN_LIST_URL["mainnet-beta"])
+			.then((response) => {
+				response.json().then((tokens) => {
+					setAllTokens(tokens);
+					setTokens(tokens);
+				});
+			}).catch((error) => {
+				console.error(error);
+			}).finally(() => {
+				setTokenLoading(false);
+			});
+	}, []);
+
+
+	const [inputTokenInfo, outputTokenInfo] = useMemo(() => {
+		return [
+			tokenMap.get(formValue.inputMint?.toBase58() || ""),
+			tokenMap.get(formValue.outputMint?.toBase58() || ""),
+		];
+	}, [formValue.inputMint?.toBase58(), formValue.outputMint?.toBase58()]);
+
+	const amountInDecimal = useMemo(() => {
+		return formValue.amount * 10 ** (inputTokenInfo?.decimals || 1);
+	}, [inputTokenInfo, formValue.amount]);
+
+	const setTokenValues = async (token: any, type?: string) => {
+		if (tokenSearchType === 'pay' || type === 'pay') {
+			setWalletBalance(token.address, "pay");
+			setFormValue((prevValue) => {
+				return {
+					...prevValue,
+					inputMint: new PublicKey(token.address),
+				};
+			});
+
+			let publicKey = wallet.publicKey || new PublicKey(token.address);
+			const info = await connection.getParsedTokenAccountsByOwner(publicKey, {
+				mint: new PublicKey(token.address),
+			});
+			const balance = info.value[0].account.data.parsed.info.tokenAmount.uiAmount;
+			let inputBalance = 0;
+			if ((balance) - 0.00001 > 0) {
+				inputBalance = (balance - 0.00001);
+			}
+			setExchangeAmount(parseFloat((inputBalance * (10 ** token?.decimals)).toString()) || 1);
+			setInputMint(new PublicKey(token.address));
+			setInputToken(token);
+		} else if (tokenSearchType === 'receive' || type === 'receive') {
+			setFormValue((prevValue) => {
+				return {
+					...prevValue,
+					outputMint: new PublicKey(token.address),
+				};
+			});
+			setWalletBalance(token.address, "receive");
+			setOutputMint(new PublicKey(token.address));
+			setOutputToken(token);
+		}
+		setShowTokenSearch(false);
+	};
+
+	const onTokenSearchChange = (value: string) => {
+		setTokenSearch(value);
+	}
+
+	useEffect(() => {
+		const options = allTokens.filter((token) => {
+			const searchLower = tokenSearch.toLowerCase();
+			return (
+				token.name.toLowerCase().includes(searchLower) ||
+				token.symbol.toLowerCase().includes(searchLower) ||
+				token.address.toLowerCase() === searchLower
+			);
+		})
+		setTokens(options);
+	}, [tokenSearch, allTokens]);
+
+	const onAggregatorSearchChange = (value: string) => {
```

Comment:
Potentially unnecessary, as it is just one expression, can call it inline in the component onChange